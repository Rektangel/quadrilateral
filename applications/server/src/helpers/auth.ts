import jwt from 'jsonwebtoken';
import { User } from '../models/User';

const { JWT_SECRET, JWT_EXPIRE_TIME } = process.env;

interface JwtPayload {
  id: string;
}

export const generateJwtToken = (user: User): string => {
  const payload: JwtPayload = {
    id: user.id,
  };

  return jwt.sign(payload, JWT_SECRET, { expiresIn: JWT_EXPIRE_TIME });
};
