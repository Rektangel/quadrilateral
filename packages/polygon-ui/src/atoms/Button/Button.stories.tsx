import * as React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';

import Button from './Button';

const ButtonContainer = styled.div`
  button:not(:last-of-type) {
    margin-right: 1rem;
  }
`;

storiesOf('Button', module)
  .addDecorator(storyFn => <ButtonContainer>{storyFn()}</ButtonContainer>)
  .add(
    'Default',
    (): JSX.Element => (
      <>
        <Button name="a">Primary</Button>
        <Button name="b" theme="secondary">
          Secondary
        </Button>
        <Button name="b" theme="success">
          Success
        </Button>
        <Button name="c" theme="danger">
          Danger
        </Button>
        <Button name="c" theme="warning">
          Warning
        </Button>
      </>
    )
  )
  .add(
    'Outlined',
    (): JSX.Element => (
      <>
        <Button name="a" outlined>
          Primary
        </Button>
        <Button name="b" theme="secondary" outlined>
          Secondary
        </Button>
        <Button name="b" theme="success" outlined>
          Success
        </Button>
        <Button name="c" theme="danger" outlined>
          Danger
        </Button>
        <Button name="c" theme="warning" outlined>
          Warning
        </Button>
      </>
    )
  )
  .add(
    'Disabled',
    (): JSX.Element => (
      <>
        <Button name="a" disabled>
          Primary
        </Button>
        <Button name="b" theme="secondary" disabled>
          Secondary
        </Button>
        <Button name="b" theme="success" disabled>
          Success
        </Button>
        <Button name="c" theme="danger" disabled>
          Danger
        </Button>
        <Button name="c" theme="warning" disabled>
          Warning
        </Button>
      </>
    )
  );
