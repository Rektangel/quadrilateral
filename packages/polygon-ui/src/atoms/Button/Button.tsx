import React from 'react';

const Button = (): JSX.Element => <button type="button">Button</button>;

export default Button;
