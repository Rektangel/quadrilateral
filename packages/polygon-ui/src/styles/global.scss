@import './abstracts/colors';

[data-theme='light'] {
  --global-background-color: #{$color-gray-100};
  --global-color: #{$color-blue-gray-800};
}

[data-theme='dark'] {
  --global-background-color: #{$color-blue-gray-800};
  --global-color: #{$color-blue-gray-100};
}

body {
  background-color: var(--global-background-color);
  color: var(--global-color);
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica,
    Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
}
