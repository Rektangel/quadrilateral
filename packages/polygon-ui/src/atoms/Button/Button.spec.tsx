import * as React from 'react';
import { renderWithTheme } from '../../helpers/testHelper';
import { dark, light } from '../../styles/themes';
import Button from './Button';

describe('Button', (): void => {
  it('with light theme renders with no errors', (): void => {
    const { asFragment } = renderWithTheme(
      <Button name="button">Light</Button>,
      light
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('with dark theme renders with no errors', (): void => {
    const { asFragment } = renderWithTheme(
      <Button name="button">Dark</Button>,
      dark
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
