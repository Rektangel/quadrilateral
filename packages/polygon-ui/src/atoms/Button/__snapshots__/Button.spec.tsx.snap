// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot with dark theme 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm glKCZb"
    name="button"
    type="button"
  >
    Dark
  </button>
</DocumentFragment>
`;

exports[`Button should match snapshot with light theme 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm gTiQGs"
    name="button"
    type="button"
  >
    Light
  </button>
</DocumentFragment>
`;
